packages/sdk-v2/src/sdk/claim/react.ts

```diff
@@ -17,10 +17,12 @@ import { noop } from "lodash";
 import { useContractFunctionWithDefaultGasFees } from "../base/hooks/useGasFees";
 
 export const useFVLink = (chainId?: number) => {
```

Comment:
how about state var initially set to null + effect running once account var changes ?

```typescript
export const useFVLink = (chainId?: number) => {
  const { account } = useEthers();
  const { chainId: defaultChainId } = useGetEnvChainId();
  const sdk = useSDK(false, "claim", chainId ?? defaultChainId) as ClaimSDK;
  const [fvLink, setFVLink] = useState(null);

  useEffect(() => {
    // skip effect if no account and sdk not initialized yet
    if (!account || !sdk) {
      return;
    }

    setFVLink(sdk.getFVLink(chainId));
  }, [account, sdk, chainId]);

  return fvLink;
};

Reply:
agreed, better not to run when unnecessary

Why the transfer to useEffect? isn't there still a utility here for useMemo, or just not 'expensive' enough?

Reply:
because 'do not do something **IF**' it's job of effects running with deps
memos are needed for 'ok let's cache something heavy to calculate while x y and z not changed'. 
Also react **does not guarantees** memos are re-calculated **only** if deps changed. Sometimes it may force memos to update. So React does not suggests to use memos for such logic

Reply:
**you rely on useMemo solely as a performance optimization. Otherwise, a [state variable](https://beta.reactjs.org/reference/react/useState#avoiding-recreating-the-initial-state) or a [ref](https://beta.reactjs.org/reference/react/useRef#avoiding-recreating-the-ref-contents) may be more appropriate**

https://beta.reactjs.org/reference/react/useMemo#caveats

Reply:
Okay got it, thanks

Reply:
Sorry, on second glance, no above solution doesn't change anything (also reverted commit)
it still runs useSDK
so we end up with the same issue as before

Reply:
the actual issue is just that the SDK is initialized before any provider has been set
and since in this use-case it is not a read-only request, there is nothing to fallback on

Reply:
also you'd need to adjust types a bit, (I used any in a few places)